app/components/network_indicator/network_indicator.js

```diff
@@ -220,9 +223,11 @@ export default class NetworkIndicator extends PureComponent {
         } = actions;
 
         if (open) {
+            this.statusUpdates = true;
             this.initializeWebSocket();
             startPeriodicStatusUpdates();
-        } else {
+        } else if (this.statusUpdates) {
+            this.statusUpdates = false;
```

Comment:
I don't understand the `statusUpdates` instance variable. It seems as if this could be `this.statusUpdates = open` in this method. Also I'm not clear on where it is used.

Reply:
So `this.statusUpdates` is basically indicating that we are polling for updates and we are only going to stop polling if this is true.

and yes I can set that to `this.statusUpdates = open` which I'm going to update

Reply:
the reason I don;t set `this.statusUpdates = open` is cause of the order of things as we only going to stop polling if we are currently doing so.

Reply:
Oh I think I get it: because `handleWebSocket` could be invoked repeatedly prior to a successful connection occurring, and the instance variable prevents unnecessary invocations of `stopPeriodicStatusUpdates` and `closeWebSocket`.